addons/account/models/account_invoice.py

```diff
@@ -444,6 +444,16 @@ def action_invoice_sent(self):
         }
 
     @api.multi
+    @api.returns('self', lambda value: value.id)
+    def message_post(self, **kwargs):
+        if self.env.context.get('mark_invoice_as_sent'):
+            for invoice in self:
+                if not invoice.sent:
+                    invoice.sent = True
```

Comment:
I would like to update invoices in batch, just for code readability: aggregate invoices, then perform a write on all invoices.